cmd/mdbd/main.go

```diff
@@ -83,6 +87,34 @@ func getSource(driverName, url string) source {
 	return source{}
 }
 
+func gracefulCleanup() {
+	os.Remove(*pidfile)
+	os.Exit(1)
+}
+
+func writePidfile() {
+	file, _ := os.Create(*pidfile)
+	defer file.Close()
+	fmt.Fprintln(file, os.Getpid())
```

Comment:
I don't think this is safe: if the pidfile cannot be created, file will be a nil pointer, so you'll end up with a panic. You need to check the err value regardless, even if you don't print it.